### ISSUE TITLE
8.8 Failing to load links

### ISSUE THREAD

**Issue by beesyrupfactory** [tonikelope/megabasterd#599]
I recently updated to v8.8 on macos sonoma using the JAR file. When copying in a link, it doesn't load and goes on forever. Downgraded to v8.4, and it works fine. I've also tried v8.7 also. So it may be possible that the logic between 8.4 -> 8.7 has changed and is preventing the links from loading.

**Comment by tonikelope** [comment]
¿Sample link?

8.7, 8.5 ?

**Comment by beesyrupfactory** [comment]
Thanks for your response Antonio @tonikelope 

I will provide the link, debug logs shortly. I will test on 8.5.
8.7 link extractor doesn't work, same as 8.8.

**Comment by beesyrupfactory** [comment]
1. Sample MEGA link: https://mega.nz/#F!gSokWLzQ!CfTNkr4pq5BbmzGLI9Y3fQ
2. 8.8
3. DEBUG logs:

`Mac OS XJava HotSpot(TM) 64-Bit Server VM 1.8.0_391 /Library/Internet Plug-Ins/JavaAppletPlugin.plugin/Contents/Home
Oct 27, 2023 7:47:00 PM com.tonikelope.megabasterd.SmartMegaProxyManager refreshSmartProxySettings
INFO: SmartProxy BAN_TIME: 300 TIMEOUT: 5 FORCE: false
Oct 27, 2023 7:47:01 PM com.tonikelope.megabasterd.SmartMegaProxyManager refreshProxyList
INFO: pool-1-thread-11 Smart Proxy Manager: proxy list refreshed (4,232)
Oct 27, 2023 7:47:02 PM com.tonikelope.megabasterd.ClipboardSpy _setEnabled
INFO: AWT-EventQueue-0 Monitoring clipboard ON...
Oct 27, 2023 7:47:04 PM com.tonikelope.megabasterd.ClipboardSpy _setEnabled
INFO: AWT-EventQueue-0 Monitoring clipboard OFF...
Exception in thread "pool-1-thread-11" java.lang.NoSuchMethodError: java.nio.file.Path.of(Ljava/lang/String;[Ljava/lang/String;)Ljava/nio/file/Path;
	at com.tonikelope.megabasterd.MegaAPI.existsCachedFolderNodes(MegaAPI.java:1055)
	at com.tonikelope.megabasterd.FolderLinkDialog._loadMegaDirTree(FolderLinkDialog.java:452)
	at com.tonikelope.megabasterd.FolderLinkDialog.lambda$new$4(FolderLinkDialog.java:103)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:750)
`

**Comment by tonikelope** [comment]
Please try 8.9

**Comment by beesyrupfactory** [comment]
Hi @tonikelope,

Thanks for your quick response. Yeah, you're absolutely right. My Java version is 1.8:

`$ java -version
java version "1.8.0_391"
Java(TM) SE Runtime Environment (build 1.8.0_391-b13)`

It's still reproducible in 8.9 using the broken link above, but I will try updating my system JDK to 11 or 14 and get back to you shortly.

**Comment by beesyrupfactory** [comment]
Fixed while using OpenJDK 21.0.1 on macOS with Homebrew, and updating the PATH variable with the updated OpenJDK version.

```shell
$ java -version
openjdk version "21.0.1" 2023-10-17
OpenJDK Runtime Environment Homebrew (build 21.0.1)
OpenJDK 64-Bit Server VM Homebrew (build 21.0.1, mixed mode, sharing)
``` 

You are absolutely right. It seems that the methods that were implemented in the latest updates were incompatible with v8.7+.

Appreciate your support @tonikelope! This can be closed 👍

**Comment by beesyrupfactory** [comment]
Oh, one suggestion!

Perhaps in the readme.md it might be worth mentioning that Java 8 is no longer supported for those running MegaBasterd via JAR file.
